src/components/organisms/UiModal/UiModal.vue

```diff
@@ -464,15 +467,37 @@ const confirmHandler = () => {
 const cancelHandler = () => {
   emit('cancel');
   emit('update:modelValue', false);
+  if (dialog.value) dialog.value.close();
 };
+
 onMounted(() => {
   if (!props.isClosable) return;
   window.addEventListener('keydown', keydownHandler);
 });
+
 onBeforeUnmount(() => {
   if (!props.isClosable) return;
   window.removeEventListener('keydown', keydownHandler);
 });
+
+watch(
+  () => props.modelValue,
+  async (value) => {
+    // NOTE: Checks if UiModal is visible, then waits for the next rendering cycle to complete using nextTick()
+    if (!value) return;
+    await nextTick();
+    if (!dialog.value) return;
+
+    // NOTE: Finds all focusable elements within the dialog, then sets focus on the first one if any are found
+    const focusableElements = dialog.value.querySelectorAll<HTMLElement>(
+      'a[href], button, textarea, input, select, [tabindex]:not([tabindex="-1"])',
+    );
+    if (focusableElements.length > 0) {
+      focusElement(focusableElements[0], true);
+    }
+  },
+  { immediate: true },
+);
```

Comment:
maybe we can add this feature to the focus trap. It knows when `isHidden` and can `getFocusableElements`

Reply:
Are you sure that in every place where we use `focus trap` we want to focus on the first focusable element? If so I can move it

Reply:
Generally yes, but we can add an option to turn it off in `focus trap`